packages/capacitor-plugin/ios/Sources/GeolocationPlugin/GeolocationPlugin.swift

```diff
@@ -120,23 +121,44 @@ private extension GeolocationPlugin {
 
     func requestLocationAuthorisation(type requestType: IONGLOCAuthorisationRequestType) {
         DispatchQueue.global(qos: .background).async {
-            self.checkIfLocationServicesAreEnabled()
+            guard self.checkIfLocationServicesAreEnabled() else { return }
             self.locationService?.requestAuthorisation(withType: requestType)
         }
     }
 
-    func checkIfLocationServicesAreEnabled() {
-        guard locationService?.areLocationServicesEnabled() ?? false else {
-            callbackManager?.sendError(.locationServicesDisabled)
-            return
+    func checkIfLocationServicesAreEnabled(_ call: CAPPluginCall? = nil) -> Bool {
```

Comment:
This may be a matter of "opinion", but we could make this method more concise like this:

```
func checkIfLocationServicesAreEnabled(_ call: CAPPluginCall? = nil) -> Bool {
    guard locationService?.areLocationServicesEnabled() == true else {
        call.map { callbackManager?.sendError($0, error: .locationServicesDisabled) } 
            ?? callbackManager?.sendError(.locationServicesDisabled)
        return false
    }
    return true
}
```

Reply:
Although I don't have a strong opinion on this, I still prefer the guard approach, as we avoid creating an unnecessary variable. But I'm ok with leaving it as is if you prefer.